.github/workflows/pre-release.yaml

```
@@ -0,0 +1,64 @@
+# This workflow concerns the preparation of the `changeset` PR and keeping it updated by tracking the changes on `main` branch.
+name: Pre-Release Preparation
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: [opened, reopened, labeled, synchronize]
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  pre-release:
+    name: Pre-Release Preparation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Check for changeset files and 'pre-release' label exist
+        id: check-conditions
+        run: |
+          if ls .changeset/*.md | grep '\.changeset\/[a-z-]\+\.md$' && [ "${{ contains(github.event.pull_request.labels.*.name, 'pre-release') }}" == "true" ]; then
+            echo "all_conditions_met=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "all_conditions_met=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      # Install Node.js and its dependencies
+      # Needed for the changeset releases
+      - name: Get NodeJS Version
+        shell: bash
+        run: |
+          NODEJS_VERSION=$(jq -r '.engines.node' "package.json")
+          echo "NODEJS_VERSION=$NODEJS_VERSION" >> $GITHUB_ENV
+
+      - name: Setup node.js and install dependencies
+        if: ${{ steps.check-conditions.outputs.all_conditions_met == 'true' }}
+        uses: ./.github/actions/nodejs-ci
+        with:
+          node-version: ${{ env.NODEJS_VERSION }}
+
+      # Keep the version of the PRs up-to-date
+      - name: Create Release Pull Request
+        id: release-pr
+        if: ${{ steps.check-conditions.outputs.all_conditions_met == 'true' }}
+        uses: changesets/action@aba318e9165b45b7948c60273e0b72fce0a64eb9 #v1.4.7
+        with:
+          branch: 'main'
+          title: 'chore: [publish-release] - bump version and update changelog'
+          commit: 'chore: bump version and update changelog'
```

```suggestion
          title: 'ci(release): bump version and update changelog'
          commit: 'ci(release): bump version and update changelog'
```
or `chore(release)` since `release:` isn't allowed